### PR TITLE
Add first draft of pinning

### PR DESCRIPTION
**Pull request by simonjbeaumont**
This has not been added to the Makefile rules yet.

### CODE REVIEW

File: planex/depend.py [hunk 1]
```diff
@@ -167,6 +149,19 @@ def main():
     for i in pkgs_to_ignore:
         print "# Will ignore: %s" % i
 
+    pins = {}
+    if args.pins_dir:
+        pins_glob = os.path.join(args.pins_dir, "*.spec")
+        pin_paths = glob.glob(pins_glob)
+        if pin_paths and args.packaging == "deb":
+            sys.stderr.write("error: Pinning not supported for debian target")
+            sys.exit(1)
```
Comment (euanh) [inline planex/depend.py]: Boo.  Could you make an 'enhancement' issue for this?

**Comment by euanh** [comment]
@simonjbeaumont: Looks great!   Could you please add some method comments to pin?   I think you have removed almost all of the code which didn't have comments, so we may be able to turn the pylint method comment check back on after this commit goes in.

**Comment by simonjbeaumont** [comment]
@euanh See additional commit, adding a docstring to all the functions in `pin.py`. Unfortunately, it seems there's a long way to go to reinstate this warning in the `pylintrc`:

```
[ssh:simonbe@morbo planex pin⚡] pylint planex 2>&1 | grep missing-docstring | wc -l
43
```

**Comment by euanh** [comment]
:collision: BOOM!  :satisfied:

**Pull request merged**